### PR TITLE
Docs install.md rebase

### LINKED ISSUE #975: document how to build freenet from source in details
I made a PR to document how to build freenet here: https://github.com/freenet/freenet-core/pull/974

ready to merge!

### PR DESCRIPTION
**Pull request by gogo2464**
fix #975 rebased.

clone of https://github.com/freenet/freenet-core/pull/974

### CODE REVIEW

**Comment by gogo2464** [comment]
ready to merge!

File: .github/workflows/build-test-format-linux.yml
Comment (iduartgomez) [inline .github/workflows/build-test-format-linux.yml:1]: This change needs to be undone, we are not changign the CI like this.  In any case we will run an additional CI step but that should not be part of this PR.

File: apps/freenet-email-app/Makefile [hunk 1]
```diff
@@ -19,6 +19,10 @@ IDENTITY_DELEGATE := $(FREENET_ROOT)$(PATH_SEP)modules$(PATH_SEP)identity-manage
 WEB_DIR_SRC := $(ROOT)$(PATH_SEP)web$(PATH_SEP)src
 FILE := $(IDENTITY_DELEGATE)$(PATH_SEP)build$(PATH_SEP)identity-manager-key.private.pem
 
+IDENTITY_DELEGATE_PRIVATE_KEY := identity-manager-key.private.pem
```
Comment (iduartgomez) [inline apps/freenet-email-app/Makefile:22]: this two variables are not being used, so why do we need them?

File: build-all.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
```
Comment (iduartgomez) [inline build-all.sh:1]: this is very specific to building the email app, so in any case move it to the freenet-email-app directory; 
those apps will be eventually move dout of this repo

File: docs/src/install.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Build freenet from source On Linux:
+
+There is a single line command to build all freenet on Linux.
```
Comment (iduartgomez) [inline docs/src/install.md:3]: This is not "building freenet", is building and running the email-app locally. Let's be more specific about that.

"If you want to run the Freenet Mail App locally you can follow this steps". This also does a bunch of things that don't have to do with that directly.

Also this is redundant with

**Review by iduartgomez — CHANGES_REQUESTED** [review]
The Cargo.lock changes have to be rolled back plus some other things.

File: build-all.sh (hunk 2 above)
Comment (gogo2464) [reply build-all.sh:1]: yes I would love. at this moment example apps rely on module folder. I do not have better implementation idea.

File: apps/freenet-email-app/Makefile (hunk 1 above)
Comment (gogo2464) [reply apps/freenet-email-app/Makefile:22]: rebase issue. sorry

File: .github/workflows/build-test-format-linux.yml
Comment (gogo2464) [reply .github/workflows/build-test-format-linux.yml:1]: the freenet email app has * like in `getrandom = { version = "*", features = ["js"] }`

Comment (gogo2464) [reply .github/workflows/build-test-format-linux.yml:1]: then I can not do better sorry

File: docs/src/install.md (hunk 3 above)
Comment (gogo2464) [reply docs/src/install.md:3]: fixed!

**Comment by gogo2464** [comment]
@iduartgomez ready to merge!

**Comment by gogo2464** [comment]
@iduartgomez ready to merge?

File: build-all.sh (hunk 2 above)
Comment (iduartgomez) [reply build-all.sh:1]: nothing stops you from moving this to:
`apps/freenet-email-app`, just use relative paths if necessary

File: .github/workflows/build-test-format-linux.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-name: CI
```
Comment (iduartgomez) [inline .github/workflows/build-test-format-linux.yml:1]: don't change this file, we are going to include a new workflow for testting the apps ina  separate PR

File: apps/freenet-email-app/Cargo.lock
Comment (iduartgomez) [inline apps/freenet-email-app/Cargo.lock:1]: so again, you shouldn't be updating the Cargo.lock yourself; it doesn't matter if there are wildcard dependencies, unless you run `cargo update` that shouldn't matter, that's what .lock files are for!

I want to be aware of all dependency version changes so I am not merging those from an unknown party and I am not going to spend time checking the diff.

Sorry to be picky about this, just delete this from the PR.

File: docs/src/SUMMARY.md [hunk 5]
```diff
@@ -14,6 +14,7 @@
 
 # Developer Guide
 
+- [Tutorial: Install freenet](install.md)
```
Comment (iduartgomez) [inline docs/src/SUMMARY.md:17]: [Tutorial: test locally a web app]

File: docs/src/install.md
Comment (iduartgomez) [inline docs/src/install.md:1]: much better! :+1:

Comment (gogo2464) [reply docs/src/install.md:1]: thanks!!! Could you fix the Cargo.lock please?

**Comment by gogo2464** [comment]
should it be ready to merge?

**Comment by gogo2464** [comment]
Hello. I permit myself to ping the freenet dev because I did not have any answer since the PR used to be mergeable. I hope there should not be any rebase conflict. 

The doc improovement seems very important!!!

File: docs/src/install.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+```
+.
+
+
+# Build freenet from source on Linux from command line:
+
+You could also run this:
+
+```bash
+wget https://sh.rustup.rs ; sh index.html -y && source "$HOME/.cargo/env" && rustup default stable && rustup target add wasm32-unknown-unknown && (sh curl -L https://git.io/n-install | bash) ; ~/n/bin/n latest ; ~/n/bin/npm install -g typescript webpack && git clone https://github.com/freenet/freenet-core/ &&    cd freenet-core &&  git submodule update --init --recursive &&         export CARGO_TARGET_DIR="$(pwd)/target" && cd stdlib/typescript/ && npm run dev.package && cd ../.. &&        cargo install --path crates/core --force &&     cargo install --path crates/fdev --force         && cd ./modules/identity-management/ && make build &&  cd ../antiflood-tokens/ && rm Cargo.lock ; make build &&      cd ../../apps/freenet-email-app && make build
```
Comment (sanity) [inline docs/src/install.md:22]: These should be on separate lines each ending with &&\

Comment (gogo2464) [reply docs/src/install.md:22]: thanks for review. I will fix it

File: apps/freenet-email-app/Cargo.lock
Comment (sanity) [reply apps/freenet-email-app/Cargo.lock:1]: @gogo2464 Have you addressed this? Is this PR still relevant? If not could you close?

**Comment by iduartgomez** [comment]
This PR seems stale.

To fix the Cargo.lock issues just delete the file and rebase on main, that should do it. Don't run `cargo update` and then you should not have this issues ion every PR.

**Comment by gogo2464** [comment]
Sorry I was over busy.

@iduartgomez is it ok if I do a rebase?

I can do it today if you want.

**Comment by al8n** [comment]
I open https://github.com/freenet/freenet-core/pull/1043 to solve the `Cargo.lock` problem and fix the comments.

**Comment by iduartgomez** [comment]
Closing cause it has been completed in #1043

**Pull request closed without merging**